Assets/Scripts/FollowPlayer.cs

```diff
@@ -13,7 +15,32 @@ void Start()
 
     void Update()
     {
+        if (_isSlowedFor <= 0)
+        {
+            _isSlowedFor = 0;
+            _slow = 0;
+        }
+        else
+        {
+            _isSlowedFor -= Time.deltaTime;
+        }
         transform.position =
-            Vector3.MoveTowards(transform.position, _player.transform.position, speed * Time.deltaTime);
+            Vector3.MoveTowards(transform.position, _player.transform.position, (speed - _slow) * Time.deltaTime);
     }
+    
+    public void ModifySpeed(float reduceSpeed, float timeSpan)
+    {
+        //avoid going over 0 speed by forcin
+        if (speed - reduceSpeed < 0)
+        {
+            _slow = speed;
```

Comment:
ohhh no, if enemy is slowed more than his speed the enemy is speed up instead xD

Reply:
as per my understanding it would walk backwards, which is definitely not wanted.